src/dune_pkg/lock_dir.ml

```diff
@@ -634,32 +634,35 @@ struct
     let open Io.O in
     Io.stats_kind lock_dir_path
     >>| function
-    | Ok S_DIR -> ()
+    | Ok S_DIR -> Ok ()
     | Error (Unix.ENOENT, _, _) ->
-      User_error.raise
-        ~hints:
-          [ Pp.concat
-              ~sep:Pp.space
-              [ Pp.text "Run"
-              ; User_message.command "dune pkg lock"
-              ; Pp.text "to generate it."
-              ]
-            |> Pp.hovbox
-          ]
-        [ Pp.textf "%s does not exist." (Path.Source.to_string lock_dir_path) ]
+      Error
+        (User_error.make
+           ~hints:
+             [ Pp.concat
+                 ~sep:Pp.space
+                 [ Pp.text "Run"
+                 ; User_message.command "dune pkg lock"
+                 ; Pp.text "to generate it."
+                 ]
+               |> Pp.hovbox
+             ]
+           [ Pp.textf "%s does not exist." (Path.Source.to_string lock_dir_path) ])
     | Error e ->
-      User_error.raise
-        [ Pp.textf "%s is not accessible" (Path.Source.to_string lock_dir_path)
-        ; Pp.textf "reason: %s" (Unix_error.Detailed.to_string_hum e)
-        ]
+      Error
+        (User_error.make
+           [ Pp.textf "%s is not accessible" (Path.Source.to_string lock_dir_path)
+           ; Pp.textf "reason: %s" (Unix_error.Detailed.to_string_hum e)
+           ])
     | _ ->
-      User_error.raise
-        [ Pp.textf "%s is not a directory." (Path.Source.to_string lock_dir_path) ]
+      Error
+        (User_error.raise
```

Comment:
Shouldn't this be `User_error.make`? Or, if we keep this as something that `raises` then without the `Error` constructor?

Reply:
Ah yep good catch, I missed that one. Fixed.